### ISSUE TITLE
Can't start existing project

### ISSUE THREAD

**Issue by justageek** [lando/lando#3630]
lando       11:26:09 ERROR ==> Error
    at /snapshot/cli/node_modules/@lando/core/lib/shell.js
From previous event:
    at Shell.sh (/snapshot/cli/node_modules/@lando/core/lib/shell.js)
    at Object.dc (/snapshot/cli/node_modules/@lando/core/lib/bootstrap.js)
    at compose (/snapshot/cli/node_modules/@lando/core/lib/bootstrap.js)
    at /snapshot/cli/node_modules/@lando/core/lib/router.js
From previous event:
    at retryEach (/snapshot/cli/node_modules/@lando/core/lib/router.js)
    at Object.start (/snapshot/cli/node_modules/@lando/core/lib/router.js)
    at /snapshot/cli/node_modules/@lando/core/lib/engine.js
    at /snapshot/cli/node_modules/@lando/core/lib/router.js
From previous event:
    at Object.eventWrapper (/snapshot/cli/node_modules/@lando/core/lib/router.js)
    at Engine.engineCmd (/snapshot/cli/node_modules/@lando/core/lib/engine.js)
    at Engine.start (/snapshot/cli/node_modules/@lando/core/lib/engine.js)
    at /snapshot/cli/node_modules/@lando/core/lib/app.js
From previous event:
    at App.start (/snapshot/cli/node_modules/@lando/core/lib/app.js)
    at Object.run (/snapshot/cli/node_modules/@lando/core/plugins/lando-core/tasks/start.js)
From previous event:
    at /snapshot/cli/lib/cli.js
    at processImmediate (node:internal/timers:466:21)
From previous event:
    at Object.handler (/snapshot/cli/lib/cli.js)
    at Object.runCommand (/snapshot/cli/node_modules/yargs/lib/command.js:238:44)
    at Object.parseArgs [as _parseArgs] (/snapshot/cli/node_modules/yargs/yargs.js:1063:30)
    at Function.get [as argv] (/snapshot/cli/node_modules/yargs/yargs.js:1004:21)
    at Cli.init (/snapshot/cli/lib/cli.js)
    at Cli.run (/snapshot/cli/lib/cli.js)
    at Object.<anonymous> (/snapshot/cli/bin/lando.js)
    at Module._compile (pkg/prelude/bootstrap.js:1926:22)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.runMain (pkg/prelude/bootstrap.js:1979:12)
    at node:internal/main/run_main_module:17:47
lando       11:26:09 VERBOSE ==> reporting error action to report=true, url=https://metrics.lando.dev, report=true, url=https://metrics.lando.dev
lando       11:26:09 DEBUG ==> reported data command=lando start, context=local, devMode=false, instance=cfde61bdb49e5a75901670dfcb017e23bc4aa7e7, nodeVersion=v16.16.0, mode=cli, type=Darwin, platform=darwin, release=22.6.0, arch=x64, product=lando, version=3.20.4, message=, stack=Error
    at /snapshot/cli/node_modules/@lando/core/lib/shell.js
From previous event:
    at Shell.sh (shell.js)
    at Object.dc (bootstrap.js)
    at compose (bootstrap.js)
    at /snapshot/cli/node_modules/@lando/core/lib/router.js
From previous event:
    at retryEach (router.js)
    at Object.start (router.js)
    at /snapshot/cli/node_modules/@lando/core/lib/engine.js
    at /snapshot/cli/node_modules/@lando/core/lib/router.js
From previous event:
    at Object.eventWrapper (router.js)
    at Engine.engineCmd (engine.js)
    at Engine.start (engine.js)
    at /snapshot/cli/node_modules/@lando/core/lib/app.js
From previous event:
    at App.start (app.js)
    at Object.run (start.js)
From previous event:
    at /snapshot/cli/lib/cli.js
    at processImmediate (timers:466:21)
From previous event:
    at Object.handler (cli.js)
    at Object.runCommand (command.js:238:44)
    at Object.parseArgs [as _parseArgs] (yargs.js:1063:30)
    at Function.get [as argv] (yargs.js:1004:21)
    at Cli.init (cli.js)
    at Cli.run (cli.js)
    at Object.<anonymous> (lando.js)
    at Module._compile (bootstrap.js:1926:22)
    at Object.Module._extensions..js (loader:1159:10)
    at Module.load (loader:981:32)
    at Function.Module._load (loader:822:12)
    at Function.runMain (bootstrap.js:1979:12)
    at node:internal/main/run_main_module:17:47, action=error, created=2023-11-27T17:26:09.792Z

**Comment by dustinleblanc** [comment]
@justageek, I don't work much on the project anymore but I saw a colleague having issues this morning and came to the issue queue to look for others with issues. Can you:

1. Post your lando config, at least what recipe you are using.
2. Can you confirm what came right _before_ the error dump? In our case my colleague appears to be having an issue installing Drupal console because the Drupal console site is stuck in a redirect loop. The site is using the Pantheon recipe.

**Comment by pvillate-rs** [comment]
I got this to work by replacing the framework config in the `.lando.yml` file. In the project it was set to `drupal8`, tried removing it entirely and it worked but drupal then started to have redirection loop issues, then found a solution for this in https://www.drupal.org/project/redirect/issues/3096948#comment-13398442, but noticed a lot of projects used that config, the workaround was not good for us.
Checked some lando examples and found out the answer, replacing `drupal8` with `drupal9` whas the answer.

**Comment by jimafisk** [comment]
I'm having similar issues. I'm trying to do a Drupal 9 to Drupal 10 upgrade. The site is currently on Drupal 9 but the config in .lando.yml says `framework: drupal8` when I first pull from Pantheon.

<hr>

If I change it to `framework: drupal10` and `lando rebuild` all the edge URLs do:
```
[502] REQUEST FAILED WITH STATUS CODE 502
```
Or
```
[503] REQUEST FAILED WITH STATUS CODE 503
```
<hr>

If I set this `framework: drupal9` the URLs all say `[301]` besides the HTTPS which says:
```
[500] REQUEST FAILED WITH STATUS CODE 500
```

I can only seem to get it to work with `framework: drupal8`, however, that seems to ship with `PHP 8.0.28` and I believe I need `PHP >=8.1.0` for Drupal 10 so it's blocking the upgrade. Any why I can't update the "framework" in the [pantheon recipe](https://docs.lando.dev/pantheon/config.html)?

**Comment by nickgeorgiou** [comment]
I simply could not get my site to start until I changed `framework: drupal10` back to  `framework: drupal9` even though I am running Drupal 10. 

```
name: **********
recipe: pantheon
config:
  framework: drupal9
  site: **********
  id: **********
```

**Comment by attheshow** [comment]
I was running into the 500 errors repeatedly on my site no matter how many times I tried to rebuild. All of the appserver URLs were saying "Request failed with status code 500". I finally loaded up the site in a browser to see what it was saying and it was reporting an issue with the PHP version being 8.1 even though my local pantheon.yml file had it specified as 8.2. I added another explicit declaration of "type: php:8.2" into the .lando.yml file under the appserver section (which theoretically shouldn't be necessary since it was already in pantheon.yml). That got everything working. Maybe the pantheon.yml file only works when using the pantheon recipe? Not sure. That sure seemed like it had always worked up until today. 🤷‍♂️😄
